File: src/ros2_bridge/mappers/foxglove_msgs/scene_entity_deletion.rs
//! Typed mapper for `foxglove_msgs/msg/SceneEntityDeletion`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn scene_entity_deletion_to_bus(msg: ros_env::foxglove_msgs::msg::SceneEntityDeletion) -> crate::foxglove_msgs::msg::v1::SceneEntityDeletion {
    crate::foxglove_msgs::msg::v1::SceneEntityDeletion {
        timestamp: Some(crate::ros2_bridge::mappers::convert::time_to_timestamp(msg.timestamp)),
        r#type: msg.type_ as i32,
        id: crate::ros2_bridge::mappers::convert::from_ros_string(msg.id),
    }
}

pub(crate) fn scene_entity_deletion_to_ros(bus: crate::foxglove_msgs::msg::v1::SceneEntityDeletion) -> ros_env::foxglove_msgs::msg::SceneEntityDeletion {
    ros_env::foxglove_msgs::msg::SceneEntityDeletion {
        timestamp: crate::ros2_bridge::mappers::convert::timestamp_to_time(bus.timestamp.unwrap_or_default()),
        type_: bus.r#type as i32,
        id: crate::ros2_bridge::mappers::convert::to_ros_string(bus.id),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsSceneEntityDeletionMapper;

impl TypedTopicMapper for FoxgloveMsgsSceneEntityDeletionMapper {
    type Ros = ros_env::foxglove_msgs::msg::SceneEntityDeletion;
    type Bus = crate::foxglove_msgs::msg::v1::SceneEntityDeletion;

    fn type_name(&self) -> &'static str {
        "foxglove_msgs/msg/SceneEntityDeletion"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(scene_entity_deletion_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(scene_entity_deletion_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/scene_entity_deletion.py
"""Generated mapper for `foxglove_msgs/msg/SceneEntityDeletion`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.foxglove_msgs.msg.v1 import SceneEntityDeletion as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def scene_entity_deletion_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.timestamp = _convert.time_to_timestamp(msg.timestamp)
    bus.type = int(msg.type)
    bus.id = str(msg.id)
    return bus


def scene_entity_deletion_to_ros(bus):
    from foxglove_msgs.msg import SceneEntityDeletion as RosMsg

    out = RosMsg()
    out.timestamp = _convert.timestamp_to_time(bus.timestamp)
    out.type = int(bus.type)
    out.id = str(bus.id)
    return out


class FoxgloveMsgsSceneEntityDeletionMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from foxglove_msgs.msg import SceneEntityDeletion as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return scene_entity_deletion_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return scene_entity_deletion_to_ros(bus)
